ark/phenotyping/create_cell_som.R

```diff
@@ -0,0 +1,81 @@
+# Trains a SOM matrix using cluster counts per cell
+
+# Usage: Rscript create_cell_som.R {fovs} {markers} {xdim} {ydim} {lr_start} {lr_end} {numPasses} {clusterCountsPath} {cellWeightsPath} {seed}
+
+# - fovs: list of fovs to cluster
+# - markers: list of channel columns to use
+# - xdim: number of x nodes to use for SOM
+# - ydim: number of y nodes to use for SOM
+# - lr_start: the start learning rate
+# - lr_end: the end learning rate
+# - numPasses: passes to make through dataset for training
+# - clusterCountsPath: path to file containing the label per pixel/meta cluster counts
+# - cellWeightsPath: path to the SOM weights file
+# - seed: the random seed to use for training
+
+library(arrow)
+library(data.table)
+library(FlowSOM)
+
+# get the command line arguments
+args <- commandArgs(trailingOnly=TRUE)
+
+# create the list of fovs
+fovs <- unlist(strsplit(args[1], split=","))
+
+# get the number of x nodes to use for the SOM
+xdim <- strtoi(args[2])
+
+# get the number of y nodes to use for the SOM
+ydim <- strtoi(args[3])
+
+# get the start learning rate
+lr_start <- as.double(args[4])
+
+# get the end learning rate
+lr_end <- as.double(args[5])
+
+# get the number of passes to make through SOM training
+numPasses <- strtoi(args[6])
+
+# get the path to the cluster counts data
+clusterCountsPath <- args[7]
+
+# get the weights write path
+cellWeightsPath <- args[8]
+
+# set the random seed
+seed <- strtoi(args[9])
+set.seed(seed)
+
+# read the cluster counts data
+print("Reading the cluster counts data for SOM training")
+clusterCountsData <- as.data.frame(arrow::read_feather(clusterCountsPath))
+
+# get the column names of the pixel clusters
+clusterCols <- colnames(clusterCountsData)[grepl(pattern="cluster_|hCluster_cap_",
+                                           colnames(clusterCountsData))]
+
+# normalized the rows by their cell size
+print("Normalizing each cell's cluster counts by cell size")
+clusterCountsNorm <- as.matrix(clusterCountsData[,clusterCols] / clusterCountsData$cell_size)
+
+# 99.9% normalize
+print("Perform 99.9% normalization")
```

Comment:
I talked with Candace and she asked that 99.9% normalization be done, so perhaps we should clarify this at standup today.

Reply:
@cliu72 What's the reason behind 99.9% normalizing the cell data?

Reply:
"Historical reasons", because it's been done in the past (Mike's favorite reason). Alex, I don't think we explicitly talked about this? If so, I might have misunderstood the question when we did. I don't think I've done a comparison to see what the effect is empirically. I can do that.